### PR TITLE
Adding scaler assignment tests for supervised trainers

### LINKED ISSUE #1790: Improve code coverage
Our project code coverage start to decrease with new features and reworked tests. We are now ~95% but were 96% recently and around ~98% the last year...

The idea is to identify the uncovered parts of code inspecting here, for example: https://codecov.io/gh/pytorch/ignite/tree/master/ignite and draft a PR with new tests.

Some of new test can be very simple and others could be very sophisticated to enter a particular code line.
Let's improve the coverage of all modules, except `distributed` for instance.

### PR DESCRIPTION
**Pull request by Moh-Yakoub**
Addresses partly #1790

Description: As part of the effort to increase cov tests, I've added few tests to test the scaler assignment of the supervised trainers method that wasn't covered in all tests:  https://codecov.io/gh/pytorch/ignite/src/ba57568cf7d4343d22171645848c4af56a0a0ae4/ignite/engine/__init__.py

Check list:

New tests are added (if a new feature is added)
New doc strings: description and/or example code are in RST format
Documentation is updated (if required)

### CODE REVIEW

**Comment by vfdev-5** [comment]
@Moh-Yakoub please do not set `Fixes #1790` as this PR does not fixes completely the issue but a part of it and just mention it as `Related to #1790` or `Addresses partly #1790`. Thanks

**Comment by Moh-Yakoub** [comment]
@vfdev-5 any thoughts on this one?

File: tests/ignite/engine/test_create_supervised.py [hunk 1]
```diff
@@ -87,6 +87,40 @@ def _test_create_supervised_trainer(
...
+        trainer = create_supervised_trainer(
+            model,
+            optimizer,
+            mse_loss,
+            device="cpu",
+            output_transform=lambda x, y, y_pred, loss: (y_pred, loss.item()),
+            amp_mode=None,
+            scaler=True,
+        )
+        assert not hasattr(trainer.state, "scaler")
```
Comment (vfdev-5) [inline tests/ignite/engine/test_create_supervised.py:121]: Which part of code we test here ?

**Review by vfdev-5 — COMMENTED** [review]
Thanks for the PR @Moh-Yakoub and sorry for delay. I have a question about the second test there...

File: tests/ignite/engine/test_create_supervised.py (hunk 1 above)
Comment (Moh-Yakoub) [reply tests/ignite/engine/test_create_supervised.py:121]: @vfdev-5 I wanted my test to be exhaustive, so I tested the two possible scenarios of the trainer state assignment, when it was set to a grad scalar and when it wasn't set.

Comment (vfdev-5) [reply tests/ignite/engine/test_create_supervised.py:121]: Yes, I understand that but the config as `amp_mode=None, scaler=True` looks very strange...

Comment (Moh-Yakoub) [reply tests/ignite/engine/test_create_supervised.py:121]: That makes sense, I will remove the second test case then

**Review by vfdev-5 — APPROVED** [review]
Thanks for the PR @Moh-Yakoub !

**Pull request merged**